enterprise/frontend/src/metabase-enterprise/caching/components/DashboardCacheSection/DashboardCacheSection.tsx

```diff
@@ -1,18 +1,53 @@
-import type { Dashboard } from "metabase-types/api";
+import { useMemo } from "react";
+import { t } from "ttag";
+import _ from "underscore";
 
-import CacheSection from "../CacheSection";
+import { useCacheConfigs } from "metabase/admin/performance/hooks/useCacheConfigs";
+import { getShortStrategyLabel } from "metabase/admin/performance/strategies";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import type { DashboardSidebarPageProps } from "metabase/dashboard/components/DashboardInfoSidebar";
+import { Flex } from "metabase/ui";
 
-interface DashboardCacheSectionProps {
-  dashboard: Dashboard;
-  onSave: (cache_ttl: number | null) => Promise<Dashboard>;
-}
+import { CacheSectionRoot } from "../CacheSection/CacheSection.styled";
+import { PolicyToken } from "../StrategyFormLauncher.styled";
+import { getDashboardId } from "../utils";
 
-const DashboardCacheSection = ({
+export const DashboardCacheSection = ({
   dashboard,
-  onSave,
-}: DashboardCacheSectionProps) => {
-  return <CacheSection initialCacheTTL={dashboard.cache_ttl} onSave={onSave} />;
-};
+  setPage,
+}: DashboardSidebarPageProps) => {
+  const dashboardId = getDashboardId(dashboard);
+
+  const { configs, loading, error } = useCacheConfigs({
+    configurableModels: ["dashboard"],
+    id: dashboardId,
+  });
+
+  const targetConfig = useMemo(
+    () => _.findWhere(configs, { model_id: dashboardId }),
+    [configs, dashboardId],
+  );
+  const savedStrategy = targetConfig?.strategy;
 
-// eslint-disable-next-line import/no-default-export -- deprecated usage
-export default DashboardCacheSection;
+  const shortStrategyLabel =
+    getShortStrategyLabel(savedStrategy, "dashboard") || t`Use default`;
```

Comment:
we probably should set the default to `Loading...` or something because this briefly flashes incorrect information.
![flashwrong](https://github.com/metabase/metabase/assets/30528226/b289c380-117f-4b75-ae41-e65cd494eb04)

Reply:
Good catch. A related problem: when you first enter the form with radio buttons, the 'Use default' radio button is briefly selected before the correct option becomes selected